### PR TITLE
DRAFT FEATURE: add 'validationErrorMessage' option

### LINKED ISSUE #3691: BUG: Regexvalidator option throws error on apply
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If I use the `validationErrorMessage` option for the `Neos.Neos/Validation/RegularExpressionValidator` on property, introduced here: https://github.com/neos/neos-ui/pull/2996.

And then try to apply a change to this property, I get an `Unsupported validation option(s) found`-error:
Screenshot_2023-06-30_at_16 43 10

### Expected Behavior

I would expect to be able to use this option without issues.

### Steps To Reproduce

Add the validationErrorMessage option to a RegularExpressionValidator, used as a property validator in the yaml NodeType definition.

Example:
```
      validation:
        'Neos.Neos/Validation/RegularExpressionValidator':
          regularExpression: /^(?:http(s)?:\/)?(\/)+[\w\-\.~:\/\?#[\]@!\$&\(\)\*\+=]*$/i
          validationErrorMessage: 'Please enter a valid relative or absolute URL.'
```

Try to apply a change to the property using this validation.
=> Error

### Environment

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by MiauzGenau**
The 'validationErrorMessage' option for Validators was introduced here: https://github.com/neos/neos-ui/pull/2996.

But is currently unusable, see: https://github.com/neos/neos-ui/issues/3691

In a first DRAFT I added 'validationErrorMessage' as an supported option, for the RegularExpressionValidator.
If this looks good, I would add the option to the other validators (adopted in [Neos-UI](https://github.com/neos/neos-ui/pull/2996)) as well. 
So please give some Feedback : )

resolved: https://github.com/neos/neos-ui/issues/3691

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked wit `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by kitsunet** [comment]
Right, this makes sense to me like this, sorry commented on the closed change.

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartV/ValidatorReference.rst [hunk 1]
```diff
@@ -487,6 +487,7 @@ Arguments
 *********
 
 * ``regularExpression`` (string): The regular expression to use for validation, used as given
+* ``validationErrorMessage`` (string): The error message to show for validation, if the regular expression validation fails
```
Comment (kitsunet) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartV/ValidatorReference.rst:490]: This file is auto generated from the code and should be updted automatically, so no need to manually change it it, that might just lead to errors with the generator. :)

File: Neos.Flow/Classes/Validation/Validator/RegularExpressionValidator.php [hunk 2]
```diff
@@ -39,7 +40,11 @@ protected function isValid($value)
     {
         $result = preg_match($this->options['regularExpression'], $value);
         if ($result === 0) {
-            $this->addError('The given subject did not match the pattern. Got: %1$s', 1221565130, [$value]);
+            if ($this->options['validationErrorMessage']) {
+                $this->addError($this->options['validationErrorMessage'], 1692962252);
```
Comment (kitsunet) [inline Neos.Flow/Classes/Validation/Validator/RegularExpressionValidator.php:44]: I wonder (mostly a question to other team members with an opinion about this) do we not want to use the same error code here? After all it's the same error just the message is different.

**Comment by mhsdesign** [comment]
As stated here: I would vouch for closing this and fixing it in the ui https://github.com/neos/flow-development-collection/pull/3134#issuecomment-1895425166

**Pull request closed without merging**